discord/audit_logs.py

```diff
@@ -238,6 +238,16 @@ def _from_data(self, data):
                     'channel': self.guild.get_channel(channel_id) or Object(id=channel_id)
                 }
                 self.extra = type('_AuditLogProxy', (), elems)()
+            elif self.action.name.endswith('pin'):
+                # the pin actions have a dict wiht some information
+                channel_id = int(self.extra['channel_id'])
+                message_id = int(self.extra['message_id'])
+                elems = {
+                    'channel': self.guild.get_channel(channel_id) or Object(id=channel_id),
+                    'message_id': message_id
```

Comment:
Possibly `self.guild._state._get_message()`?

Reply:
Don't think so.
Audit logs aren't sent via WS events so there's no way to know when exactly to make a cache of the message.